### ISSUE TITLE
jitouch won't work after installation.

### ISSUE THREAD

**Issue by idhun90** [aaronkollasch/jitouch#21]
<img width="559" alt="2022-05-11_02-16-59" src="https://user-images.githubusercontent.com/51053410/167686723-c01b9f68-77cd-4933-bb47-c3c2588e4549.png">

Hello. Long time no see.
I'm using it well, but my acquaintance sent me a screenshot saying it won't work after installation. (M1, Air)

1. To give Jitouch accessibility access -> go to "Security and Privacy" -> "Privacy" -> "Accessibility". There should be a list with "Allow the apps below to control your computer." and Jitouch should be in it.
Click the lock to make changes
Check the box next to Jitouch
If Jitouch isn't on the list:

Run open ~/Library/PreferencePanes/Jitouch.prefPane/Contents/Resources or open /Library/PreferencePanes/Jitouch.prefPane/Contents/Resources to open a Finder window that should contain a Jitouch application.
Drag Jitouch to the Accessibility list.

2. I also entered the code below at the terminal.

xattr -d com.apple.quarantine Jitouch.prefPane

But it was still not solved, so I asked for help.

**Comment by idhun90** [comment]
> 

Configuration error

Unable to load the Jitouch configuration panel.

Yes, using Install-Jitouch.pkg

**Comment by aaronkollasch** [comment]
What version of macOS are they using?

**Comment by idhun90** [comment]
> What version of macOS are they using?

It is said that the update is just in progress. It's hard to check right now. I expect 12.3 version of m1 MacBook Air that he bought yesterday.

**Comment by aaronkollasch** [comment]
Ok, that's surprising as Jitouch should be compatible, and the package is signed so there shouldn't be a permissions error. 
Just to check for a quarantine flag, can they run `xattr -lr /Library/PreferencePanes/Jitouch.prefPane` if they installed for all users or `xattr -lr ~/Library/PreferencePanes/Jitouch.prefPane` if they installed for a single user?

**Comment by idhun90** [comment]
> If either shows a `com.apple.quarantine` flag, it can be removed with `sudo xattr -rd com.apple.quarantine /Library/PreferencePanes/Jitouch.prefPane` or `xattr -rd com.apple.quarantine ~/Library/PreferencePanes/Jitouch.prefPane`

After the update, I will remotely remove the Jitouch and try to install it again.

Can you recommend the installation order?

1. To give Jitouch accessibility access
2. Enter commands from the terminal

or 

1. Enter commands from the terminal
2. To give Jitouch accessibility access

**Comment by aaronkollasch** [comment]
After installing, first try opening the System Preferences panel without running any commands. If you see the error message you posted, try running the commands (first the `-lr` command to check if there are any flags, which there shouldn't be). Also, perhaps a restart will fix it?

The accessibility access isn't needed to open the PrefPane, so don't worry about it yet.

**Comment by idhun90** [comment]
> After installing, first try opening the System Preferences panel without running any commands. If you see the error message you posted, try running the commands (first the `-lr` command to check if there are any flags, which there shouldn't be). Also, perhaps a restart will fix it?
> 
> The accessibility access isn't needed to open the PrefPane, so don't worry about it yet.

Thank you.

First, I will test and if the same error occurs, I will use the four commands that you gave us.

**Comment by aaronkollasch** [comment]
Sounds good. To be clear, you will only need to run one of each pair of commands, depending on where you install the PrefPane. The ones containing `~/Library/...` are if you installed "for this user", and `/Library/...` are if you installed "for all users".

You should avoid installing Jitouch.prefPane in both `/Library/PreferencePanes` and `~/Library/PreferencePanes` as you can only allow one in Accessibility prefs, but that's for later.

**Comment by idhun90** [comment]
> Sounds good. To be clear, you will only need to run one of each pair of commands, depending on where you install the PrefPane. The ones containing `~/Library/...` are if you installed "for this user", and `/Library/...` are if you installed "for all users".
> 
> You should avoid installing Jitouch.prefPane in both `/Library/PreferencePanes` and `~/Library/PreferencePanes` as you can only allow one in Accessibility prefs, but that's for later.

Thank you for letting me know clearly.

Then I'll proceed with the method below.

1. install (for this user)
2. try opening the System Preferences panel without running any commands.
3. If I see the error message I posted -> xattr -lr ~/Library/PreferencePanes/Jitouch.prefPane
4. If shows a com.apple.quarantine flag -> xattr -rd com.apple.quarantine ~/Library/PreferencePanes/Jitouch.prefPane

but I don't know what is 'If shows a com.apple.quarantine flag'?

**Comment by aaronkollasch** [comment]
Looks good. If some files are flagged for quarantine, the output from `xattr -lr ...` will have some lines with a filename and `com.apple.quarantine`. Otherwise, it should be empty.

**Comment by idhun90** [comment]
> Looks good. If some files are flagged for quarantine, the output from `xattr -lr ...` will have some lines with a filename and `com.apple.quarantine`. Otherwise, it should be empty.

Thank you.

The final printout must be empty to finish it properly, right?

**Comment by aaronkollasch** [comment]
If it's not empty, can you share the flags you're seeing? They probably don't matter, but it might be good to see what they are as the next command removes them.

**Comment by idhun90** [comment]
> If it's not empty, can you share the flags you're seeing? They probably don't matter, but it might be good to see what they are as the next command removes them.

I can't do it right now because my friend is asleep. I'm going to proceed tomorrow during the day.😂

I will test it and let you know the information about the results.
Thank you for your help.

**Comment by idhun90** [comment]
> > If it's not empty, can you share the flags you're seeing? They probably don't matter, but it might be good to see what they are as the next command removes them.
> 
> I can't do it right now because my friend is asleep. I'm going to proceed tomorrow during the day.😂
> 
> I will test it and let you know the information about the results. Thank you for your help.

Installation as a pkg file does not appear in the System Configuration panel.🥲

**Comment by idhun90** [comment]
> > > If it's not empty, can you share the flags you're seeing? They probably don't matter, but it might be good to see what they are as the next command removes them.
> > 
> > 
> > I can't do it right now because my friend is asleep. I'm going to proceed tomorrow during the day.😂
> > I will test it and let you know the information about the results. Thank you for your help.
> 
> Installation as a pkg file does not appear in the System Configuration panel.🥲

Resolved after reboot.
Thank you!

**Comment by idhun90** [comment]
> Awesome! Should I close the issue then?

Yes, thank you for your help :)
